wgpu-hal/src/dynamic/command.rs

```diff
@@ -1,8 +1,29 @@
-use crate::{BufferBinding, CommandEncoder};
+use crate::{
+    BufferBarrier, BufferBinding, BufferCopy, CommandEncoder, DeviceError, Label, MemoryRange,
+};
 
-use super::DynBuffer;
+use super::{DynBuffer, DynResourceExt as _};
+
+pub trait DynCommandEncoder: std::fmt::Debug {
```

Comment:
issue: We don't document a safety contract or adherence thereto for these methods. We should do this.